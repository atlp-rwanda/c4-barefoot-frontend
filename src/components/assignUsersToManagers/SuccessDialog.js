import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

export default function SuccessDialog(props) {
  let [open, setOpen] = React.useState(props.open);
  return (
    <Dialog
      open={open}
      onClose={()=>{setOpen(!open)}}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title" style={{color: 'green'}}>Success!</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description" style={{color: '#219653'}}>
          The users have been assigned to respective managers successfully!
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={()=>{setOpen(false)}} color="primary">
          OK
        </Button>
      </DialogActions>
    </Dialog>
  )
}
