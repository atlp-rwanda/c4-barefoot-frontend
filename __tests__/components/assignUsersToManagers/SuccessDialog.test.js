import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SuccessDialog from '../../../src/components/assignUsersToManagers/SuccessDialog';

describe('<SuccessDialog />', ()=> {
    let wrapper;
    const props = { open: true };
    it( 'It should render the SuccessDialog component', ()=>{
        wrapper= shallow(<SuccessDialog {...props}/>)
        const component= toJson(wrapper);
        expect(component).toMatchSnapshot();
    });
});
