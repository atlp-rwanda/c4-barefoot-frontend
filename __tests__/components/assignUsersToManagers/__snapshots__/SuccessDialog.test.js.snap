// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessDialog /> It should render the SuccessDialog component 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="alert-dialog-title"
    style={
      Object {
        "color": "green",
      }
    }
  >
    Success!
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))
      id="alert-dialog-description"
      style={
        Object {
          "color": "#219653",
        }
      }
    >
      The users have been assigned to respective managers successfully!
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      onClick={[Function]}
    >
      OK
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
